cmd/upgrade/roles/cmd.go

```diff
@@ -0,0 +1,788 @@
+		r.Reporter.Infof("Starting to upgrade the operator IAM roles and policies")
+	}
+
+	operatorRolePolicies, err := r.OCMClient.GetPolicies("OperatorRole")
+	if err != nil {
+		r.Reporter.Errorf("Expected a valid role creation mode: %s", err)
+		os.Exit(1)
+	}
+
+	if isOperatorPolicyUpgradeNeeded {
+		err = upgradeOperatorPolicies(
+			args.isInvokedFromClusterUpgrade,
+			mode,
+			r,
+			isUpgradeNeedForAccountRolePolicies,
+			operatorRolePolicies,
+			env,
+			policyVersion,
+			credRequests,
+			cluster,
+			operatorRolePolicyPrefix,
+		)
+		if err != nil {
+			r.Reporter.Errorf("%s", err)
+			os.Exit(1)
+		}
+	}
+
+	if len(missingRolesInCS) > 0 {
+		for _, operator := range missingRolesInCS {
+			roleName := roles.GetOperatorRoleName(cluster, operator)
+			exists, _, err := r.AWSClient.CheckRoleExists(roleName)
+			if err != nil {
+				return r.Reporter.Errorf("Error when detecting checking missing operator IAM roles %s", err)
+			}
+			if !exists {
+				err = createOperatorRole(
+					mode,
+					r,
+					cluster,
+					missingRolesInCS,
+					args.isInvokedFromClusterUpgrade,
+					operatorRolePolicies,
+					operatorRolePath,
+					operatorPolicyPath,
+					operatorRolePolicyPrefix,
+				)
+				if err != nil {
+					r.Reporter.Errorf("%s", err)
+					os.Exit(1)
+				}
+			}
+		}
+	}
+	return err
+}
+
+func LogError(key string, ocmClient *ocm.Client, defaultPolicyVersion string, err error, reporter *rprtr.Object) {
+	reporter.Debugf("Logging throttle error")
+	if strings.Contains(err.Error(), "Throttling") {
+		ocmClient.LogEvent(key, map[string]string{
+			ocm.Response:   ocm.Failure,
+			ocm.Version:    defaultPolicyVersion,
+			ocm.IsThrottle: "true",
+		})
+	}
+}
+
+func upgradeAccountRolePoliciesFromCluster(
+	reporter *rprtr.Object,
+	awsClient aws.Client,
+	cluster *v1.Cluster,
+	accountID string,
+	policies map[string]string,
+	policyVersion string,
+) error {
+	for file, role := range aws.AccountRoles {
+		roleName, err := aws.GetAccountRoleName(cluster, role.Name)
+		if err != nil {
+			return err
+		}
+		if roleName == "" {
+			reporter.Debugf("Cluster '%s' does not include expected role '%s'", cluster.ID(), role.Name)
+			continue
+		}
+		prefix, err := aws.GetPrefixFromAccountRole(cluster, role.Name)
+		if err != nil {
+			return err
+		}
+		promptString := fmt.Sprintf("Upgrade the '%s' role policy latest version (%s) ?", roleName, policyVersion)
+		if !confirm.Prompt(true, promptString) {
+			if args.isInvokedFromClusterUpgrade {
+				return reporter.Errorf("Account roles need to be upgraded to proceed" +
+					"")
+			}
+			continue
+		}
+		filename := fmt.Sprintf("sts_%s_permission_policy", file)
+		policiesDetails, err := awsClient.GetAttachedPolicy(&roleName)
+		if err != nil {
+			return err
+		}
+
+		if aws.CountAttachedPoliciesInDetails(policiesDetails) > 1 {
+			return weberr.Errorf("More than one attached policy per account role not supported. %s", roleName)
```

Comment:
This will not work for  breakout policies and with managed policies coming in i don't think can enforce this